src/parser_v3.py

```diff
@@ -0,0 +1,617 @@
+        state_name = entry['detectedstate']
+        try:
+            state = STATE_CODES[state_name]
+        except KeyError:
+            # Entries having unrecognized state names are discarded
+            if state_name:
+                print('[{}: {}] [{}] [Bad state: {}] {}'.format(
+                    i, j + 2, entry['dateannounced'], state_name,
+                    entry['numcases']))
+            continue
+
+        try:
+            fdate = datetime.strptime(entry['dateannounced'], '%d/%m/%Y')
+            date = datetime.strftime(fdate, '%Y-%m-%d')
+        except ValueError:
+            # Bad date
+            print('[{}: {}] [Bad date: {}] {}: {} {}'.format(
+                i, j + 2, entry['dateannounced'], entry['detectedstate'],
+                entry['detecteddistrict'], entry['numcases']))
+            continue
+
+        district = entry['detecteddistrict']
+        if not district:
+            district = 'Unassigned'
+        elif district.lower() in DISTRICTS_DICT[state]:
+            district = DISTRICTS_DICT[state][district.lower()]
+        elif district.lower() in DISTRICTS_ADDITIONAL:
+            district = DISTRICTS_ADDITIONAL[district.lower()]
+        else:
+            # Print unexpected district names
+            print('[{}: {}] [{}] [Unexpected district: {}] {}'.format(
+                i, j + 2, date, district, state))
+
+        try:
+            count = int(entry['numcases'])
+        except ValueError:
+            print('[{}: {}] [{}] [Bad numcases: {}] {}: {}'.format(
+                i, j + 2, date, entry['numcases'], state, district))
+            continue
+
+        if entry['currentstatus'] == 'Hospitalized' or i < 3:
+            inc(data[date]['TT']['delta'], 'confirmed', count)
+            inc(data[date][state]['delta'], 'confirmed', count)
+            # Don't parse old district data since it's unreliable
+            if i > 2:
+                inc(data[date][state]['districts'][district]['delta'],
+                    'confirmed', count)
+        elif entry['currentstatus'] == 'Recovered':
+            inc(data[date]['TT']['delta'], 'recovered', count)
+            inc(data[date][state]['delta'], 'recovered', count)
+            # Don't parse old district data unreliable since it's unreliable
+            if i > 2:
+                inc(data[date][state]['districts'][district]['delta'],
+                    'recovered', count)
+        elif entry['currentstatus'] == 'Deceased':
+            inc(data[date]['TT']['delta'], 'deceased', count)
+            inc(data[date][state]['delta'], 'deceased', count)
+            # Don't parse old district data unreliable since it's unreliable
+            if i > 2:
+                inc(data[date][state]['districts'][district]['delta'],
+                    'deceased', count)
+
+
+def parse_outcome(outcome_data, i):
+    for j, entry in enumerate(outcome_data['deaths_recoveries']):
+        state_name = entry['state']
```

Comment:
Will there be an issue if the state is in all caps?

Reply:
Yes, but I haven't handled that since I'm assuming that states are selected from a drop-down in the sheet. I'll add a check if that's not the case.